tensorflow/core/kernels/fused_batch_norm_op.cc

```diff
@@ -71,7 +188,13 @@ struct FusedBatchNorm<CPUDevice, T, U> {
                   Tensor* y_output, Tensor* batch_mean_output,
                   Tensor* batch_var_output, Tensor* saved_mean_output,
                   Tensor* saved_var_output, TensorFormat tensor_format,
-                  bool is_training) {
+                  ScratchAllocator* reserve_space_allocator,
+                  ScratchAllocator* workspace_allocator, bool is_training) {
+    if (reserve_space_allocator != nullptr) {
+      Tensor* dummy_reserve_space = nullptr;
```

Comment:
I saw your comment below, please also add the same comment here.
Also, where is this space used in the code?

My understanding from reading the code is, it's used if and only if `CUDNN_VERSION >= 7402` **and** `is_training == true`. Also, it's needed only in `FusedBatchNormOpV3::Compute()`, then why not just use `allocate_temp()` instead of `allocate_output()`?

Reply:
The `reserved_space` will also be used in back-propagation. It stores some intermediate results from forward computation and will be used in back-propagation for better performance.
My understanding the space created by `allocate_temp()` can be only alive as long as that op.